source/Ship.cpp

```diff
@@ -3550,10 +3550,13 @@ int Ship::TakeDamage(vector<Visual> &visuals, const DamageDealt &damage, const G
 	else if(heat < .9 * MaximumHeat())
 		isOverheated = false;
 
+	double scale = Energy() * 220.;
+	double jamChance = ionization > .1 ? min(0.5, scale ? ionization / scale : 1.) : 0.;
```

Comment:
Is this the jam chance equation? It's not great to reproduce that here. If it is currently only present where it is used and inaccessible to any other code, perhaps it should be accessed, either as a member method of Ship or a namespace method that you pass a ship reference/pointer to.

Reply:
As I said in the description I'll open a PR to have it only be used once.

Currently its literally the same code in anti missile and normal weapon firing run as many times as you have guns shooting per frames

Reply:
I made it a namespace function, cant resist the chance to make the code more elegant if you ask for it x)